plugins/hint_calls/hint_calls.py

```diff
@@ -371,52 +373,37 @@ def get_custom_viewer_hint(self, view, place):
                 return None
 
             curline = idaapi.get_custom_viewer_curline(view, True)
-            _, x, y = idaapi.get_custom_viewer_place(view, True)
+            
+            # sometimes get_custom_viewer_place() returns [x, y] and sometimes [place_t, x, y].
+            # we want the place_t.
+            viewer_place = idaapi.get_custom_viewer_place(view, True)
+            if len(viewer_place) != 3:
+                return None
+
+            _, x, y = viewer_place
             ea = place.toea()
 
             # "color" is a bit of misnomer: its the type of the symbol currently hinted
             color = get_color_at_char(curline, x)
             if color != idaapi.COLOR_ADDR:
                 return None
 
-            # for COLOR_ADDR tokens, we get something like:
-            #   401000sub_401000
-            # so we will need to prune the address from the start before we can use it :-(
-            token = get_token_at_char(curline, x)
-
-            # enumerate the operands of the instruction at this address
-            # and search the token for the operand text
-            func_name = None
-            for i in range(3):
-                o = idc.GetOpnd(ea, i)
-                if not o:
-                    break
-
-                # if we have `offset sub_401000`, we want: `sub_401000`
-                if ' ' in o:
-                    o = o.partition(' ')[2]
-
-                if o in token:
-                    func_name = o
-                    break
-
-            if not func_name:
+            # PATCHED: instead grab the FAR references to code (not necessarilty a branch/call/jump by itself)
+            far_code_references = [xref.to for xref in idautils.XrefsFrom(ea, ida_xref.XREF_FAR) if idc.isCode(idc.GetFlags(xref.to))]
```

Comment:
i think this solution is good enough. its not as precise as doing all the crazy parsing as above, but i think it will work just as well in practice. since we're not using `get_token_at_char`, then we can also remove the embedded `idapython_lex_curline` at the start of the file. the simplicity might trump the rigor.

thoughts?